vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow/src/test/java/com/vaadin/flow/component/datetimepicker/DateTimePickerBinderValidationTest.java

```diff
@@ -0,0 +1,126 @@
+package com.vaadin.flow.component.datetimepicker;
+
+import com.vaadin.flow.data.binder.Binder;
+import com.vaadin.flow.data.binder.BindingValidationStatus;
+import com.vaadin.flow.data.binder.BindingValidationStatusHandler;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import java.time.LocalDateTime;
+
+public class DateTimePickerBinderValidationTest {
+
+    private static final String BINDER_FAIL_MESSAGE = "YEAR_LESS_THAN_MIN";
+    private static final String REQUIRED_MESSAGE = "REQUIRED";
+
+    @Captor
+    private ArgumentCaptor<BindingValidationStatus<?>> statusCaptor;
+
+    @Mock
+    private BindingValidationStatusHandler statusHandlerMock;
+
+    public static class Bean {
+        private LocalDateTime date;
+
+        public LocalDateTime getDate() {
+            return date;
+        }
+
+        public void setDate(LocalDateTime date) {
+            this.date = date;
+        }
+    }
+
+    @Before
+    public void init() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void elementWithConstraints_componentValidationNotMet_elementValidationFails() {
```

Comment:
In theory, all the components now will rely on the client side validation. Like `min`/`max` will be checked on a `NumberField` instance first on the client and then on the server. So, that would make sense to add the ITs. At the other hand, the server side validation for the same constraints should work on the same way as the client side, so it may be beneficial to have unit tests to check them.

One thing it got me thinking is how the client side validation status will play along with the server side check. For instance, the developer sets `max` to `NumberField` and the user inserts a number greater than `max`. The client side validation will show it as invalid and so the event will send this information to the server. When `checkValidity` on the server is called, will the method say that the component is invalid based on the client status or will it check the constraints again and only rely on the client side status when all server side constraints pass?